torch/csrc/autograd/python_variable.cpp

```diff
@@ -53,6 +53,32 @@ static const char* VOLATILE_WARNING =
     "volatile was removed and now has no effect. Use "
     "`with torch.no_grad():` instead.";
 
+#ifdef USE_DEPLOY
+// used only in libtorch_deployinterpreter.so
+// there are muliple copies of the python interpreter that
+// can shared Tensors, so rather than use their internal pointer
+// to a PyObject use a library-local map.
+static std::unordered_map<void*, PyObject*> impl_to_pyobj;
```

Comment:
I would have a couple questions about this :) 
- Why is this only done for Tensor? The same thing is true for all the python bindings we have no (for autograd, thinking about Node in particular)?
- Why is this not implemented in `variable.cpp` in the implementation of torch::autograd::impl::pyobj? That would feel better to have all the definition of these functions at the same place.

Reply:
1. AFAIK, other python bindings do not use internal function pointers. Does Node do this? I know that storage doesn't, as well as all the stuff using Pybind. If so, then we need similar code for those. Autograd is not supported in this patch for other subtler reasons inside the engine code itself that assumes a single python, so fixes to Node can come later.
2. The code needs to be defined in `libtorch_python.so`, the python bindings, not in `libtorch_cpu.so`. There is only one copy of `libtorch_cpu.so` so putting a hash table in there would not give us independent tables per interpreter. Technically, we can move everything into `python_variable.cpp` if that is more clear, but then it further exposes the internal details of TensorImpl.

Reply:
Thanks for the clarifications!
For future reference, the Node does the same thing here: https://github.com/pytorch/pytorch/blob/fb07aca7b05edfb9d41bca950425e70d0b29bcbf/torch/csrc/autograd/function.h#L294-L303

Reply:
I haven't noticed in any meaningful tests, including timing JIT doing nothing against Python doing nothing, where you would expect this to show up. However, I am not confident in my tests here to say that we should do this universally. Using pybind11 requires buying into other infrastructure (e.g. std::shared_ptr containers for the objects), so it is not ideal here.